### PR TITLE
add decsription for rnnblock

### PR DESCRIPTION
**Pull request by jiangnanhugo**
fix https://github.com/Theano/Theano/issues/5109

### CODE REVIEW

**Comment by jiangnanhugo** [comment]
add description for `theano.gpuarray.dnn.RNNBlock`

**Comment by lamblin** [comment]
flake8 tests report the following errors:
```
/home/travis/build/Theano/Theano/theano/gpuarray/dnn.py:2456:1: W293 blank line contains whitespace
/home/travis/build/Theano/Theano/theano/gpuarray/dnn.py:2460:1: W293 blank line contains whitespace
```

File: theano/gpuarray/dnn.py [hunk 1]
```diff
@@ -2443,17 +2443,28 @@ class RNNBlock(object):
...
+        lstm: A four-gate Long Short-Term Memory network with no peephole connections.
+        gru: A three-gate network consisting of Gated Recurrent Units.
     input_mode : {'linear', 'skip'}
         linear: input will be multiplied by a biased matrix
         skip: No operation is performed on the input.  The size must match the hidden size.
     direction_mode : {'unidirectional', 'bidirectional'}
-        unidirectional: The network operates recurrently from the
-                        first input to the last.
-
+        unidirectional: The network operates recurrently from thefirst input to the last.
```
Comment (abergeron) [inline theano/gpuarray/dnn.py]: thefirst -> the first

Comment (jiangnanhugo) [inline theano/gpuarray/dnn.py]: got it. thanks.

File: theano/gpuarray/dnn.py [hunk 2]
```diff
@@ -2443,17 +2443,28 @@ class RNNBlock(object):
...
+        number of the recurrent layer you want to set.
     rnn_mode : {'rnn_relu', 'rnn_tanh', 'lstm', 'gru'}
-        See cudnn documentation for ``cudnnRNNMode_t``.
+        rnn_relu: A single-gate recurrent neural network with a ReLU activation function.
 
+        .. math::
+
+        h_t=ReLU(W_ix_t+U_ih_{t-1}+b_{wi}+b_{Ri})
+        rnn_tanh: A single-gate recurrent neural network with a tanh activation function.
+
```
Comment (abergeron) [inline theano/gpuarray/dnn.py]: There is whitespace here.

File: theano/gpuarray/dnn.py [hunk 3]
```diff
@@ -2443,17 +2443,28 @@ class RNNBlock(object):
...
 
+        .. math::
+
+        h_t=ReLU(W_ix_t+U_ih_{t-1}+b_{wi}+b_{Ri})
+        rnn_tanh: A single-gate recurrent neural network with a tanh activation function.
+        
+        .. math::
+
+        h_t=tanh(W_ix_t+U_ih_{t-1}+b_{wi}+b_{Ri})
+
```
Comment (abergeron) [inline theano/gpuarray/dnn.py]: And here

**Comment by nouiz** [comment]
jenkins test this

**Comment by nouiz** [comment]
thanks

**Pull request merged**